pkg/latte/node_change_password.go

```diff
@@ -0,0 +1,95 @@
+package latte
+
+import (
+	"context"
+	"errors"
+
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+)
+
+func init() {
+	workflow.RegisterNode(&NodeChangePassword{})
+}
+
+type NodeChangePassword struct {
+	UserID            string             `json:"user_id,omitempty"`
+	AuthenticatorKind authenticator.Kind `json:"authenticator_kind,omitempty"`
+}
+
+func (n *NodeChangePassword) Kind() string {
+	return "latte.NodeChangePassword"
+}
+
+func (n *NodeChangePassword) GetEffects(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) (effs []workflow.Effect, err error) {
+	return nil, nil
+}
+
+func (n *NodeChangePassword) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	return []workflow.Input{
+		&InputTakeChangePassword{},
+	}, nil
+}
+
+func (n *NodeChangePassword) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	var inputTakeChangePassword inputTakeChangePassword
+	switch {
+	case workflow.AsInput(input, &inputTakeChangePassword):
+		info, err := n.getPasswordAuthenticator(deps)
+		// The user doesn't have the password authenticator
+		// always returns invalid credentials error
+		if errors.Is(err, api.ErrNoAuthenticator) {
+			return nil, api.ErrInvalidCredentials
```

Comment:
1. Move `ChangePasswordFailed` and `ErrNoPassword` in pkg/lib/interaction/nodes/change_password.go to pkg/api/errors.go
2. Return `ErrNoPassword` when the user has no password